docs/shared/guides/nx-and-ts.md

```diff
@@ -0,0 +1,151 @@
+```
+
+Alternatively, if you have already an existing tsc based library/application, you can run the `@nrwl/js:convert-to-swc` generator to migrate the package from TSC to SWC.
+
+The following command converts the `hello-tsc` library to SWC:
+
+```bash
+nx generate @nrwl/js:convert-to-swc --name=hello-tsc
+```
+
+## Publish your TypeScript packages to NPM
+
+Nx doesn’t provide an out of the box process for the publishing itself and leaves it to the developer to invoke the final command. The reason is that the actual publishing process can be very specific to your project and target you deploy to and might have a lot of custom pre-deployment setup (e.g. generating changelogs, determining semver etc.). Make sure to check out our [community page](/community) as there are a lot of community provided packages integrating into the publishing process.
+
+However, integrating your custom publishing process with Nx can be pretty straightforward, especially with the help of Nx [run-commands](/{{framework}}/executors/run-commands-builder) and [“Target Dependencies”](/{{framework}}/core-concepts/configuration#target-dependencies).
+
+To add a new run-command to our project, we can leverage the `run-command` generator:
+
+```bash
+nx g @nrwl/workspace:run-commands publish --project hello-tsc --command 'npm publish --tag=latest --access public'
+```
+
+This adds a new target `publish` to our existing `project.json` configuration of the `hello-tsc` project. Let's also make sure to adjust the `cwd` of our command, which should be in the output folder of the `hello-tsc` package.
+
+```json
+{
+  "root": "packages/hello-tsc",
+  "sourceRoot": "packages/hello-tsc/src",
+  "projectType": "library",
+  "targets": {
+    "build": { ... },
+    "lint": { ... },
+    "test": { ... },
+    "publish": {
+      "executor": "@nrwl/workspace:run-commands",
+      "outputs": [],
+      "options": {
+        "command": "npm publish --tag=latest --access public",
+        "cwd": "dist/packages/hello-tsc"
+      }
+    }
+  },
+  "tags": []
+}
+```
+
+With that, we're already set up. By running the following commands we can publish our package to NPM.
+
+```bash
+nx build hello-tsc
+nx publish hello-tsc
+```
+
+We can however even automate this further by leveraging the `targetDependencies` feature. There's clearly a dependency from `publish -> build`, in that we want to make sure the `build` happens before the `publish` command, since the latter depends on the output of the former. We can define such relationship in the `nx.json`:
+
+```json
+{
+  ...
+  "targetDependencies": {
+    ...
+    "publish": [
+      {
+        "target": "build",
+        "projects": "self"
+      }
+    ]
+  },
+  ...
+}
+```
+
+Now, just running `nx publish hello-tsc` will automatically run the `nx build hello-tsc` command first. And of course, if `build` has already run, it won't execute again, thanks to [Nx computation caching](/{{framework}}/core-extended/computation-caching).
```

Comment:
I'd a section at the end. If you want to generate JS apps and libs, pass --js flag.

Reply:
added a note at the beginning, so people know there's the option